### ISSUE TITLE
java.lang.IllegalArgumentException occurred after deleting the entire project

### ISSUE THREAD

**Issue by scruel** [hsz/idea-gitignore#477]
### Prerequisites

* [x] Plugin is in the latest version
* [x] Issue was not reported yet
* [x] Stack trace (if provided) contains `mobi.hsz.idea.gitignore` package name

### Description

_java.lang.IllegalArgumentException occurred after deleting the entire project_

```
Argument for @NotNull parameter 'directory' of mobi/hsz/idea/gitignore/util/Utils.isUnder must not be null
java.lang.IllegalArgumentException: Argument for @NotNull parameter 'directory' of mobi/hsz/idea/gitignore/util/Utils.isUnder must not be null
	at mobi.hsz.idea.gitignore.util.Utils.$$$reportNull$$$0(Utils.java)
	at mobi.hsz.idea.gitignore.util.Utils.isUnder(Utils.java)
	at mobi.hsz.idea.gitignore.IgnoreManager.isFileIgnored(IgnoreManager.java:318)
	at mobi.hsz.idea.gitignore.daemon.IgnoredEditingNotificationProvider.createNotificationPanel(IgnoredEditingNotificationProvider.java:103)
	at mobi.hsz.idea.gitignore.daemon.IgnoredEditingNotificationProvider.createNotificationPanel(IgnoredEditingNotificationProvider.java:47)
	at com.intellij.ui.EditorNotificationsImpl$3.performInReadAction(EditorNotificationsImpl.java:148)
	at com.intellij.openapi.progress.util.ReadTask.a(ReadTask.java:66)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:47)
	at com.intellij.openapi.progress.util.ReadTask.runBackgroundProcess(ReadTask.java:66)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.a(ProgressIndicatorUtils.java:243)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:170)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:548)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:493)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.a(ProgressIndicatorUtils.java:241)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.access$000(ProgressIndicatorUtils.java:47)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils$4.run(ProgressIndicatorUtils.java:192)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


java.lang.IllegalArgumentException: Argument for @NotNull parameter 'directory' of mobi/hsz/idea/gitignore/util/Utils.isUnder must not be null
	at mobi.hsz.idea.gitignore.util.Utils.$$$reportNull$$$0(Utils.java)
	at mobi.hsz.idea.gitignore.util.Utils.isUnder(Utils.java)
	at mobi.hsz.idea.gitignore.IgnoreManager.isFileIgnored(IgnoreManager.java:318)
	at mobi.hsz.idea.gitignore.vcs.IgnoreFileStatusProvider.getFileStatus(IgnoreFileStatusProvider.java:70)
	at com.intellij.openapi.vcs.impl.FileStatusManagerImpl.calcStatus(FileStatusManagerImpl.java:135)
	at com.intellij.openapi.vcs.impl.FileStatusManagerImpl.getStatus(FileStatusManagerImpl.java:248)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.getFileColor(FileEditorManagerImpl.java:368)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters.updateFileColor(EditorsSplitters.java:378)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.b(FileEditorManagerImpl.java:399)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.access$1500(FileEditorManagerImpl.java:111)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$MyFileStatusListener.a(FileEditorManagerImpl.java:1780)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$MyFileStatusListener.b(FileEditorManagerImpl.java:1766)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:314)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:416)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:821)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:649)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)


java.lang.IllegalArgumentException: Argument for @NotNull parameter 'directory' of mobi/hsz/idea/gitignore/util/Utils.isUnder must not be null
	at mobi.hsz.idea.gitignore.util.Utils.$$$reportNull$$$0(Utils.java)
	at mobi.hsz.idea.gitignore.util.Utils.isUnder(Utils.java)
	at mobi.hsz.idea.gitignore.IgnoreManager.isFileIgnored(IgnoreManager.java:318)
	at mobi.hsz.idea.gitignore.vcs.IgnoreFileStatusProvider.getFileStatus(IgnoreFileStatusProvider.java:70)
	at com.intellij.openapi.vcs.impl.FileStatusManagerImpl.calcStatus(FileStatusManagerImpl.java:135)
	at com.intellij.openapi.vcs.impl.FileStatusManagerImpl.getStatus(FileStatusManagerImpl.java:248)
	at com.intellij.ide.navigationToolbar.NavBarPresentation.getTextAttributes(NavBarPresentation.java:113)
	at com.intellij.ide.navigationToolbar.NavBarPanel.isRebuildUiNeeded(NavBarPanel.java:361)
	at com.intellij.ide.navigationToolbar.NavBarUpdateQueue.rebuildUi(NavBarUpdateQueue.java:183)
	at com.intellij.ide.navigationToolbar.NavBarUpdateQueue$2.after(NavBarUpdateQueue.java:176)
	at com.intellij.ide.navigationToolbar.NavBarUpdateQueue$AfterModelUpdate.run(NavBarUpdateQueue.java:285)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:337)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:327)
	at com.intellij.util.ui.update.MergingUpdateQueue.lambda$flush$1(MergingUpdateQueue.java:277)
	at com.intellij.util.ui.UIUtil.invokeAndWaitIfNeeded(UIUtil.java:2781)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:288)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:246)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:246)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:417)
	at com.intellij.util.Alarm$Request.access$700(Alarm.java:344)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:384)
	at com.intellij.util.Alarm$Request.run(Alarm.java:395)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:237)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:821)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:649)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

### Steps to Reproduce

1. delete another entire project
2. exception occurred.

**Expected behavior:** _What you expect to happen_

**Actual behavior:** _What actually happens_

**Reproduces how often:** _What percentage of the time does it reproduce?_

### Versions

**Plugin:** 

```
Can be found in Settings > Plugins > .ignore
```

**IDE:** 

```
IntelliJ IDEA 2017.2.4
Build #IU-172.4155.36, built on September 12, 2017
Licensed to scruel
Subscription is active until August 27, 2018
For educational use only.
JRE: 1.8.0_152-release-915-b11 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
```

**OS:** 

```
Windows 10
```

### Additional Information

_Any additional information, configuration or data that might be necessary to reproduce the issue._

**Comment by hsz** [comment]
@scruel can you please answer for all questions from the issue template?

**Comment by scruel** [comment]
@hsz sorry, done.

**Comment by hsz** [comment]
@scruel  Thank you for this report.

It's fixed  (3dfdbe4) and will be released with v2.2.2

**Comment by hsz** [comment]
Fix has been published with `v2.2.2`.
